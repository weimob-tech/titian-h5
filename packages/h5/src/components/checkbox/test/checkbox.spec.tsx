import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { TestContainer } from '../../common/test/container';
import { TiCheckbox } from '../index';

describe('渲染 checkbox', () => {
  it('基本渲染', async () => {
    const page = await newSpecPage({
      components: [TiCheckbox, TestContainer],
      template: () => (
        <test-container>
          <ti-checkbox label="单选框" />
          <ti-checkbox>单选框</ti-checkbox>
          <ti-checkbox />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('默认选中状态', async () => {
    const page = await newSpecPage({
      components: [TiCheckbox, TestContainer],
      template: () => (
        <test-container>
          <ti-checkbox label="默认选中" checked />
          <ti-checkbox label="默认选中" defaultChecked />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('修改多选框的圆角度数', async () => {
    const page = await newSpecPage({
      components: [TiCheckbox],
      template: () => (
        <test-container>
          <ti-checkbox label="方型" shape="square" />
          <ti-checkbox label="圆型" shape="circle" />
          <ti-checkbox label="自定义角度" shape={4} />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('修改多选框的颜色', async () => {
    const page = await newSpecPage({
      components: [TiCheckbox, TestContainer],
      template: () => (
        <test-container>
          <ti-checkbox label="方型" color="red" />
          <ti-checkbox label="方型" color="#ff0000" />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('禁用多选框点击效果', async () => {
    const page = await newSpecPage({
      components: [TiCheckbox, TestContainer],
      template: () => (
        <test-container>
          <ti-checkbox label="禁用全部点击效果" disabled />
          <ti-checkbox label="禁用文字点击效果" label-disabled />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('组件图标设置', async () => {
    const page = await newSpecPage({
      components: [TiCheckbox],
      template: () => (
        <test-container>
          <ti-checkbox label="默认图标" />
          <ti-checkbox label="自定义图标大小" size={64} />
          <ti-checkbox label="自定义图标 plus" icon="plus" />
          <ti-checkbox label="自定义图标 false" icon="false" />
          <ti-checkbox label="自定义图标 slot" slot="">
            <div slot="icon">icon</div>
          </ti-checkbox>
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('组件图标设置', async () => {
    const page = await newSpecPage({
      components: [TiCheckbox],
      template: () => <ti-checkbox label="自定义图标 false" icon={false} />,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('监听 全部 点击事件', async () => {
    let checked = false;
    const change = jest.fn((e: CustomEvent) => {
      checked = e.detail;
    });
    const page = await newSpecPage({
      components: [TiCheckbox],
      template: () => <ti-checkbox label="单选框" onChange={change} />,
    });

    {
      // label 点击
      const label = page.root.shadowRoot.querySelector<HTMLDivElement>('.titian-checkbox-label');
      label.click();
      expect(checked).toBeTruthy();
    }

    {
      // icon 点击
      const iconWrap = page.root.shadowRoot.querySelector<HTMLDivElement>('.titian-checkbox-icon-wrap');
      iconWrap.click();
      expect(checked).toBeFalsy();
    }
  });

  it('禁用全部点击事件', async () => {
    const change = jest.fn();

    const page = await newSpecPage({
      components: [TiCheckbox],
      template: () => <ti-checkbox disabled onChange={change} label="单选框" />,
    });

    {
      // label 点击
      const label = page.root.shadowRoot.querySelector<HTMLDivElement>('.titian-checkbox-label');
      label.click();
      expect(change).toBeCalledTimes(0);
    }

    {
      // icon 点击
      const iconWrap = page.root.shadowRoot.querySelector<HTMLDivElement>('.titian-checkbox-icon-wrap');
      iconWrap.click();
      expect(change).toBeCalledTimes(0);
    }
  });

  it('禁用 label 点击事件', async () => {
    const change = jest.fn();
    const page = await newSpecPage({
      components: [TiCheckbox],
      template: () => <ti-checkbox labelDisabled onChange={change} label="单选框" />,
    });
    {
      // label 点击
      const label = page.root.shadowRoot.querySelector<HTMLDivElement>('.titian-checkbox-label');
      label.click();
      expect(change).toBeCalledTimes(0);
    }

    {
      // icon 点击
      const iconWrap = page.root.shadowRoot.querySelector<HTMLDivElement>('.titian-checkbox-icon-wrap');
      iconWrap.click();
      expect(change).toBeCalledTimes(1);
    }
  });
});
