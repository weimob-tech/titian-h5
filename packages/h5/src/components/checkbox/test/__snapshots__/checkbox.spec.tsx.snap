// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`渲染 checkbox 修改多选框的圆角度数 1`] = `
<ti-checkbox>
  <mock:shadow-root>
    <div class="titian-checkbox titian-checkbox-horizontal">
      <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: -999px;">
        <div class="titian-checkbox-icon-wrap">
          <ti-icon ext-class="titian-checkbox-icon titian-checkbox-icon-zoom-out" name="checkbox-hollow" size="32" style="width: 16px; height: 16px;"></ti-icon>
        </div>
      </div>
      <div aria-hidden="true" class="titian-checkbox-label">
        方型
      </div>
    </div>
  </mock:shadow-root>
</ti-checkbox>
`;

exports[`渲染 checkbox 修改多选框的颜色 1`] = `
<test-container>
  <!---->
  <ti-checkbox>
    <mock:shadow-root>
      <div class="titian-checkbox titian-checkbox-horizontal">
        <div aria-hidden="true" class="titian-checkbox-icon-content" style="--checkbox-icon-color: red; --base-radius-size: 999px;">
          <div class="titian-checkbox-icon-wrap">
            <ti-icon ext-class="titian-checkbox-icon titian-checkbox-icon-zoom-out" name="checkbox-hollow" size="32" style="width: 16px; height: 16px;"></ti-icon>
          </div>
        </div>
        <div aria-hidden="true" class="titian-checkbox-label">
          方型
        </div>
      </div>
    </mock:shadow-root>
  </ti-checkbox>
  <ti-checkbox>
    <mock:shadow-root>
      <div class="titian-checkbox titian-checkbox-horizontal">
        <div aria-hidden="true" class="titian-checkbox-icon-content" style="--checkbox-icon-color: #ff0000; --base-radius-size: 999px;">
          <div class="titian-checkbox-icon-wrap">
            <ti-icon ext-class="titian-checkbox-icon titian-checkbox-icon-zoom-out" name="checkbox-hollow" size="32" style="width: 16px; height: 16px;"></ti-icon>
          </div>
        </div>
        <div aria-hidden="true" class="titian-checkbox-label">
          方型
        </div>
      </div>
    </mock:shadow-root>
  </ti-checkbox>
</test-container>
`;

exports[`渲染 checkbox 基本渲染 1`] = `
<test-container>
  <!---->
  <ti-checkbox>
    <mock:shadow-root>
      <div class="titian-checkbox titian-checkbox-horizontal">
        <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 999px;">
          <div class="titian-checkbox-icon-wrap">
            <ti-icon ext-class="titian-checkbox-icon titian-checkbox-icon-zoom-out" name="checkbox-hollow" size="32" style="width: 16px; height: 16px;"></ti-icon>
          </div>
        </div>
        <div aria-hidden="true" class="titian-checkbox-label">
          单选框
        </div>
      </div>
    </mock:shadow-root>
  </ti-checkbox>
  <ti-checkbox>
    <mock:shadow-root>
      <div class="titian-checkbox titian-checkbox-horizontal">
        <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 999px;">
          <div class="titian-checkbox-icon-wrap">
            <ti-icon ext-class="titian-checkbox-icon titian-checkbox-icon-zoom-out" name="checkbox-hollow" size="32" style="width: 16px; height: 16px;"></ti-icon>
          </div>
        </div>
        <div aria-hidden="true" class="titian-checkbox-label titian-checkbox-label-empty">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
    单选框
  </ti-checkbox>
  <ti-checkbox>
    <mock:shadow-root>
      <div class="titian-checkbox titian-checkbox-horizontal">
        <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 999px;">
          <div class="titian-checkbox-icon-wrap">
            <ti-icon ext-class="titian-checkbox-icon titian-checkbox-icon-zoom-out" name="checkbox-hollow" size="32" style="width: 16px; height: 16px;"></ti-icon>
          </div>
        </div>
        <div aria-hidden="true" class="titian-checkbox-label titian-checkbox-label-empty">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-checkbox>
</test-container>
`;

exports[`渲染 checkbox 禁用多选框点击效果 1`] = `
<test-container>
  <!---->
  <ti-checkbox>
    <mock:shadow-root>
      <div class="titian-checkbox titian-checkbox-disabled titian-checkbox-horizontal">
        <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 999px;">
          <div class="titian-checkbox-icon-wrap">
            <ti-icon ext-class="titian-checkbox-icon titian-checkbox-icon-" name="checkbox-hollow" size="32" style="width: 16px; height: 16px;"></ti-icon>
          </div>
        </div>
        <div aria-hidden="true" class="titian-checkbox-label">
          禁用全部点击效果
        </div>
      </div>
    </mock:shadow-root>
  </ti-checkbox>
  <ti-checkbox label-disabled="">
    <mock:shadow-root>
      <div class="titian-checkbox titian-checkbox-horizontal">
        <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 999px;">
          <div class="titian-checkbox-icon-wrap">
            <ti-icon ext-class="titian-checkbox-icon titian-checkbox-icon-zoom-out" name="checkbox-hollow" size="32" style="width: 16px; height: 16px;"></ti-icon>
          </div>
        </div>
        <div aria-hidden="true" class="titian-checkbox-label">
          禁用文字点击效果
        </div>
      </div>
    </mock:shadow-root>
  </ti-checkbox>
</test-container>
`;

exports[`渲染 checkbox 组件图标设置 1`] = `
<ti-checkbox>
  <mock:shadow-root>
    <div class="titian-checkbox titian-checkbox-horizontal">
      <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 999px;">
        <div class="titian-checkbox-icon-wrap">
          <ti-icon ext-class="titian-checkbox-icon titian-checkbox-icon-zoom-out" name="checkbox-hollow" size="32" style="width: 16px; height: 16px;"></ti-icon>
        </div>
      </div>
      <div aria-hidden="true" class="titian-checkbox-label">
        默认图标
      </div>
    </div>
  </mock:shadow-root>
</ti-checkbox>
`;

exports[`渲染 checkbox 组件图标设置 2`] = `
<ti-checkbox>
  <mock:shadow-root>
    <div class="titian-checkbox titian-checkbox-horizontal">
      <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 999px;">
        <div class="titian-checkbox-icon-wrap">
          <slot name="icon"></slot>
        </div>
      </div>
      <div aria-hidden="true" class="titian-checkbox-label">
        自定义图标 false
      </div>
    </div>
  </mock:shadow-root>
</ti-checkbox>
`;

exports[`渲染 checkbox 默认选中状态 1`] = `
<test-container>
  <!---->
  <ti-checkbox>
    <mock:shadow-root>
      <div class="titian-checkbox titian-checkbox-checked titian-checkbox-horizontal">
        <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 999px;">
          <div class="titian-checkbox-icon-wrap titian-checkbox-icon-wrap-checked">
            <ti-icon ext-class="titian-checkbox-icon titian-checkbox-icon-zoom-in titian-checkbox-icon-checked" name="checkbox-hollow" size="32" style="width: 16px; height: 16px;"></ti-icon>
          </div>
        </div>
        <div aria-hidden="true" class="titian-checkbox-label">
          默认选中
        </div>
      </div>
    </mock:shadow-root>
  </ti-checkbox>
  <ti-checkbox>
    <mock:shadow-root>
      <div class="titian-checkbox titian-checkbox-checked titian-checkbox-horizontal">
        <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 999px;">
          <div class="titian-checkbox-icon-wrap titian-checkbox-icon-wrap-checked">
            <ti-icon ext-class="titian-checkbox-icon titian-checkbox-icon-zoom-in titian-checkbox-icon-checked" name="checkbox-hollow" size="32" style="width: 16px; height: 16px;"></ti-icon>
          </div>
        </div>
        <div aria-hidden="true" class="titian-checkbox-label">
          默认选中
        </div>
      </div>
    </mock:shadow-root>
  </ti-checkbox>
</test-container>
`;
